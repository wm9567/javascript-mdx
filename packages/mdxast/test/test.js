const fs = require('fs')
const unified = require('unified')
const remark = require('remark-parse')
const rehype = require('remark-rehype')
const html = require('rehype-stringify')

const toMdx = require('..')

const fixture = fs.readFileSync('test/fixture.md', 'utf8')

const parseFixture = str => {
  const parser = unified()
    .use(remark)
    .use(toMdx)
    .use(rehype)
    .use(html)

  return parser.processSync(str).contents
}

test('it parses a file', () => {
  parseFixture(fixture)
})
