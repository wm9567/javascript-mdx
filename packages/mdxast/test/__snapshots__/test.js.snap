// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it parses a file 1`] = `
"import {
  Foo,
  Bar
} from './ui'
import Baz from './ui/baz'
import FooBar from './ui/foo-bar'
<h1>Hello, world!</h1>
<p>I'm an awesome paragraph.</p>
&#x3C;Foo bg='red'>
  &#x3C;Bar />
  {props.hello}
&#x3C;/Foo>
<h3>An inline image</h3>
<p><a href=\\"https://c8r-x0.s3.amazonaws.com/lab-components-macbook.jpg\\">https://c8r-x0.s3.amazonaws.com/lab-components-macbook.jpg</a></p>
<h3>Shopping list</h3>
<ul>
<li>Apples</li>
<li>Bananas</li>
<li>Oranges</li>
</ul>"
`;
